pkg/alert/alertdeployment.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright (C) 2018 Synopsys, Inc.
+
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements. See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownershia. The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied. See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package alert
+
+import (
+	"fmt"
+	"strconv"
+
+	horizonapi "github.com/blackducksoftware/horizon/pkg/api"
+	"github.com/blackducksoftware/horizon/pkg/components"
+)
+
+// getAlertDeployment returns a new deployment for an Alert
+func (a *SpecConfig) getAlertDeployment() *components.Deployment {
+	replicas := int32(1)
+	deployment := components.NewDeployment(horizonapi.DeploymentConfig{
+		Replicas:  &replicas,
+		Name:      "alert",
+		Namespace: a.config.Namespace,
+	})
+	deployment.AddMatchLabelsSelectors(map[string]string{"app": "alert", "tier": "alert"})
+
+	pod := a.getAlertPod()
+
+	deployment.AddPod(pod)
+
+	deployment.AddLabels(map[string]string{"app": "alert"})
+	return deployment
+}
+
+// getAlertPod returns a new Pod for an Alert
+func (a *SpecConfig) getAlertPod() *components.Pod {
+	pod := components.NewPod(horizonapi.PodConfig{
+		Name: "alert",
+	})
+	pod.AddLabels(map[string]string{"app": "alert", "tier": "alert"})
+
+	pod.AddContainer(a.getAlertContainer())
+
+	vol := a.getAlertVolume()
+
+	pod.AddVolume(vol)
+
+	pod.AddLabels(map[string]string{"app": "alert"})
```

Comment:
I maybe a bit confused about this, but why the `AddLabels` twice for the same key `app`?

Reply:
Nice catch!